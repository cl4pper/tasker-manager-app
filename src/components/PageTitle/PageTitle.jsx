import React from 'react';
import classNames from 'classnames';

// STYLE
import './PageTitle.scss';

function PageTitle(props) {
	const { children, light, subtitle } = props;

	return (
		<div
			className={classNames('PageTitle', {
				'PageTitle--dark': !light,
				'PageTitle--light': light,
			})}>
			<h1 className="PageTitle__title">{children}</h1>
			{subtitle ? <p className="PageTitle__subtitle">{subtitle}</p> : null}
		</div>
	);
}

export { PageTitle };
