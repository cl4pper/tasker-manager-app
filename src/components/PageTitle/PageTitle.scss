.PageTitle {

  @media (max-width: 500px) {
    text-align: center;
  }

  &__title {
    @include fontTitle("lg");
    letter-spacing: 0;
    font-weight: bold;
    margin: 0;
    display: flex;
    align-items: center;
    color: inherit;
  }

  &__subtitle {
    @include fontText("sm");
    margin: 0;
    color: inherit;
  }

  &--dark {
    color: $text-color-primary;
  }

  &--light {
    color: $text-color-secondary;
  }
}
