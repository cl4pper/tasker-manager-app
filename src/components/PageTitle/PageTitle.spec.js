import React from 'react';
import renderer from 'react-test-renderer';
import { PageTitle } from './PageTitle';

describe('PageTitle', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<PageTitle children="Hello" light={false} subtitle="World" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
