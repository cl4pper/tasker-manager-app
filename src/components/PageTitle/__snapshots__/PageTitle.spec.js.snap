// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle snapshot renders 1`] = `
<div
  className="PageTitle PageTitle--dark"
>
  <h1
    className="PageTitle__title"
  >
    Hello
  </h1>
  <p
    className="PageTitle__subtitle"
  >
    World
  </p>
</div>
`;
